File: ex137.py
class Solution:
    def singleNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """

        return int((sum(set(nums)) * 3 - sum(nums)) / 2)

sol = Solution()
print(sol.singleNumber([0, 1, 0, 1, 0, 1, 99]))

File: code/ex137.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        int once = 0, twice = 0;
        for (int num : nums) {
            once = ~twice & (once ^ num);
            twice = ~once & (twice ^ num);
        }
        return once;
    }
};